### PR TITLE
Move generic documentation to shared location

### PR DESCRIPTION
**Pull request by adrienbernede**
This PR moves and complete documentation to a shared location (versioned in https://github.com/LLNL/radiuss-ci).

The parts of the documentation concerned are the usage of Uberenv and Gitlab CI.

Relates to https://github.com/LLNL/Umpire/pull/442 & https://github.com/LLNL/RAJA/pull/905

### CODE REVIEW

File: docs/sphinx/developer/uberenv.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+MacOS case
+^^^^^^^^^^
+
+In CHAI, the Spack configuration for MacOS contains the default compilers depending on the OS version (`compilers.yaml`), and a commented section to illustrate how to add `CMake` as an external package. You may install CMake with homebrew, for example.
+
+
+Using Uberenv to generate the host-config file
+----------------------------------------------
+
+We have seen that we can safely use `gcc@8.1.0` on quartz. Let us ask for the default configuration first, and then produce static libs, have OpenMP support and run the benchmarks:
```
Comment (adrienbernede) [inline docs/sphinx/developer/uberenv.rst:63]:
```suggestion
We have seen that we can safely use `gcc@8.1.0` on quartz. Let us ask for the default configuration first, and then ask for RAJA support and link to develop version of RAJA:
```

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
@davidbeckingsale Same here, I think this should be merged sooner than later :)

**Comment by davidbeckingsale** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**